### PR TITLE
Improve dashboard and marketplace connection

### PR DESCRIPTION
**Pull request by lkostrowski**
I want to merge this change because...

It integrates with Marketplace, adding new navigation items and changing how routes between Dashboard and Marketplace are resolved

I also disables Marketplace preview to avoid broken Marketplace on production. Since this is a preview we agreed with @timuric to hide it for few days.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,14 @@ Create `.env` file in a root directory or set environment variables with followi
 - `STATIC_URL` - URL where the static files are located.
   E.g. if you use S3 bucket, you should set it to the bucket's URL. By default Saleor assumes you serve static files from the root of your site at `http://localhost:9000/`.
 
+- `MARKETPLACE_URL`  - URL where Marketplace App can is located, if not found, will not render navigation link to Marketplace
+
+- `SALEOR_APPS_PAGE_PATH` - Path joined to `MARKETPLACE_URL` to render Saleor Apps page
+ 
+- `SALEOR_APPS_JSON_PATH` - Path joined to `MARKETPLACE_URL` to fetch list of Saleor Apps as JSON
+-
```
Comment (krzysztofwolski) [inline README.md:89]:
```suggestion
 
```

File: src/components/AppLayout/menuStructure.ts [hunk 2]
```diff
@@ -75,9 +75,16 @@ function useMenuStructure(
           },
           {
             ariaLabel: "marketplace",
-            label: intl.formatMessage(sectionNames.marketplace),
-            id: "marketplace",
-            url: marketplaceUrl,
+            // label: intl.formatMessage(sectionNames.marketplace),
```
Comment (krzysztofwolski) [inline src/components/AppLayout/menuStructure.ts:78]:
```suggestion
```

File: src/config.ts [hunk 3]
```diff
@@ -8,8 +8,13 @@ export const getAppMountUri = () =>
...
 export const IS_CLOUD_INSTANCE = process.env.IS_CLOUD_INSTANCE === "true";
-export const MARKETPLACE_URL = process.env.MARKETPLACE_URL;
-export const SALEOR_APPS_ENDPOINT = process.env.SALEOR_APPS_ENDPOINT;
+export const MARKETPLACE_URL = window.__SALEOR_CONFIG__.MARKETPLACE_URL;
+export const MARKETPLACE_SALEOR_APPS_PAGE_PATH =
+  window.__SALEOR_CONFIG__.SALEOR_APPS_PAGE_PATH;
+export const MARKETPLACE_SALEOR_APPS_JSON_PATH =
+  window.__SALEOR_CONFIG__.SALEOR_APPS_JSON_PATH;
+export const MARKETPLACE_APP_TEMPLATE_GALLERY_PATH =
+  window.__SALEOR_CONFIG__.APP_TEMPLATE_GALLERY_PATH;
```
Comment (andrzejewsky) [inline src/config.ts:17]: **suggestion**:  how about creating functions just like we have with getApiUrl? (consistency)

**Review by andrzejewsky — DISMISSED** [review]

File: src/config.ts (hunk 3 above)
Comment (lkostrowski) [reply src/config.ts:17]: I think there should be 1 layer with raw ENVs, second layer with some functions to resolve them

However, Dashboard is not my scope so I suggest you plan conventions and refactors with the team. When I contribute, I follow style I see in the codebase

File: src/apps/components/AppsListPage/AppsListPage.tsx [hunk 4]
```diff
@@ -179,7 +198,9 @@ const AppsListPage: React.FC<AppsListPageProps> = ({
       <AppPageTabs
         showSaleorApps={saleorAppsEnabled}
         className={styles.topTabs}
-        onChange={setActiveTab}
+        onChange={v => {
+          updatePath(v);
+        }}
```
Comment (krzysztofwolski) [inline src/apps/components/AppsListPage/AppsListPage.tsx:203]:
```suggestion
        onChange={updatePath}
```

**Review by krzysztofwolski — APPROVED** [review]

**Review by orzechdev — DISMISSED** [review]

**Review by krzysztofwolski — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Comment by lkostrowski** [comment]
Hey @NyanKiyoshi can you take a look?

Since contract between Dashboard and Marketplace is changing, I'd like to turn it off on production (its on preview mode), finalize this on staging and then plan to merge production

File: README.md [hunk 5]
```diff
@@ -81,6 +81,14 @@ Create `.env` file in a root directory or set environment variables with followi
 - `STATIC_URL` - URL where the static files are located.
   E.g. if you use S3 bucket, you should set it to the bucket's URL. By default Saleor assumes you serve static files from the root of your site at `http://localhost:9000/`.
 
+- `MARKETPLACE_URL`  - URL where Marketplace App can is located, if not found, will not render navigation link to Marketplace
+
+- `SALEOR_APPS_PAGE_PATH` - Path joined to `MARKETPLACE_URL` to render Saleor Apps page
+ 
+- `SALEOR_APPS_JSON_PATH` - Path joined to `MARKETPLACE_URL` to fetch list of Saleor Apps as JSON
+
+- `APP_TEMPLATE_GALLERY_PATH` - Path joined to `MARKETPLACE_URL` to render App Template Gallery page
```
Comment (NyanKiyoshi) [inline README.md:90]: UX: mark those are optional so the users know they can skip safely

**Review by NyanKiyoshi — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (lkostrowski) [reply README.md:90]: 👍  will add this later

**Pull request merged**